subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -210,5 +210,56 @@ If you use the `@Input` annotation on a file-based property, Gradle wouldn't con
 To fix this problem, you need to tell Gradle if the file property represents an input file, in which case you should annotate it with `@InputFile`, or a directory, in which case it should be annotated with `@InputDirectory`.
 If what you really wanted to say is that the actual file path is an input, then you should return a `String` instead which corresponds to the absolute path of the file.
 
+[[unresolvable_input]]
+== Unresolvable input
 
+This error indicates that a file collection is used as an input of a task, but that resolution failed.
+In general, this shouldn't happen.
+However, when this happens it means that Gradle cannot properly determine the task dependencies.
 
+As a workaround, you can declare an explicit task dependency instead using `Task#dependsOn`.
+
+[[implicit_dependency]]
+== Implicit dependencies between tasks
+
+This error indicates that you have a task which depends on another, but that no explicit or implicit dependency is declared between those two tasks.
+As a consequence, the results of the build are dependent on the order of execution of tasks, often referred to "accidental dependencies between tasks".
+Often, this is because you refer directly to the output file of another task instead of using the task directly as an input.
+
+For example, imagine that you have a task which takes a `ConfigurableFileCollection` as an input and that you have declared a dependency on the `jar` task using this:
+
+```
+someTask {
+    inputFile.from(jar.archivePath)
+}
+```
+
+The `jar.archivePath` property is of type `File`, which doesn't carry any task dependency.
+It means that if you call `someTask` _after_ `jar` has been called, the task will succeed, but if the jar is removed, for example, the task would fail.
+
+To fix this, you can declare a `Property` as an input instead:
+
+```
+someTask {
+    inputFile.from(jar.archiveFile)
+}
+```
+
+The `jar.archiveFile` property is of type `Provider<RegularFile>` which properly carries task dependencies: Gradle will be able to know that the file is generated by the `jar` task.
+
+It's actually even easier to add an implicit dependency to the task itself:
```

Comment:
I wouldn't mention this. We don't want to encourage this and actually get rid of the output file collection in the long run.

Reply:
This is actually orthogonal and probably one of the features I like most today (that you can declare a dependency on a task without caring). I don't think we should remove this. However, we can change the implementation. This is what we do in dependency resolution for artifact dependencies: we can refer to an artifact as a task, as long as it has a _single_ declared output. In most cases that works really well.